linera-chain/src/manager.rs

```diff
@@ -224,7 +224,12 @@ impl ChainManager {
         );
         if let Some(validated) = validated {
             ensure!(
-                validated.value().is_validated(),
+                validated.value().is_validated()
+                    && validated
+                        .value()
+                        .executed_block()
+                        .map(|executed_block| &executed_block.block)
+                        == Some(new_block),
```

Comment:
Was this a bug ?

Reply:
Mmm ok, more like a performance bug apparently.

Reply:
Before oracles, it was only a performance issue, yes.

Now it's important, since if there is a validated block, we don't re-execute the block proposal and use the outcome (including oracle values) from the validated block.

But you just made me realize that this isn't even enough: We'll have to [make the block proposer _sign_ not only the proposed block](https://github.com/linera-io/linera-protocol/blob/eb0c79aabaab5a6e973db51f66b18ee31a1b5361/linera-chain/src/data_types.rs#L795), but also the included validated block.
Otherwise, if there were already multiple rounds where the same proposed block got validated with different oracle values, someone who isn't the current round leader could take my proposal signature and swap out the validated block for a different one (same block, different oracle values) and make my round fail.

And when we're fixing that we might as well also address the duplication that required this equality here in the first place: Instead of including the validated block certificate, only include the _lite_ certificate plus the block execution outcome, and then verify that together with the proposed block, the latter matches the lite certificate's hash.